packages/server/src/api/routes/tests/search.spec.ts

```diff
@@ -76,13 +91,190 @@ describe.each([
+      { query: { range: { age: { low: 0, high: 11 } } }, expected: rows },
+    ]
+
+    it.each(numberSearchTests)(
+      `should be able to run query: $query`,
+      async ({ query, expected }) => {
+        const savedRows = await Promise.all(
+          rows.map(r => config.api.row.save(table._id!, r))
+        )
+        const { rows: foundRows } = await config.api.row.search(table._id!, {
+          tableId: table._id!,
+          query,
+        })
+        expect(foundRows).toEqual(
+          expect.arrayContaining(
+            expected.map(r =>
+              expect.objectContaining(savedRows.find(sr => sr.age === r.age)!)
+            )
+          )
+        )
+      }
+    )
+  })
+
+  describe("dates", () => {
+    beforeEach(async () => {
+      table = await config.api.table.save(
+        tableForDatasource(datasource, {
+          schema: {
+            dob: {
+              name: "dob",
+              type: FieldType.DATETIME,
+            },
+          },
+        })
+      )
+    })
+
+    const rows = [
+      { dob: new Date("2020-01-01") },
+      { dob: new Date("2020-01-10") },
+    ]
+
+    interface DateSearchTest {
+      query: SearchFilters
+      expected: (typeof rows)[number][]
+    }
+
+    const dateSearchTests: DateSearchTest[] = [
+      { query: {}, expected: rows },
+      {
+        query: { onEmptyFilter: EmptyFilterOption.RETURN_ALL },
+        expected: rows,
+      },
+      {
+        query: { onEmptyFilter: EmptyFilterOption.RETURN_NONE },
+        expected: [],
+      },
+      {
+        query: { equal: { dob: new Date("2020-01-01") } },
+        expected: [rows[0]],
+      },
+      { query: { equal: { dob: new Date("2020-01-02") } }, expected: [] },
+      {
+        query: { notEqual: { dob: new Date("2020-01-01") } },
+        expected: [rows[1]],
+      },
+      {
+        query: { oneOf: { dob: [new Date("2020-01-01")] } },
+        expected: [rows[0]],
+      },
+      {
+        query: {
+          range: {
+            dob: {
+              low: new Date("2020-01-01").toISOString(),
+              high: new Date("2020-01-05").toISOString(),
+            },
+          },
+        },
+        expected: [rows[0]],
+      },
+      {
+        query: {
+          range: {
+            dob: {
+              low: new Date("2020-01-01").toISOString(),
+              high: new Date("2020-01-10").toISOString(),
+            },
+          },
+        },
+        expected: rows,
+      },
+      {
+        query: {
+          range: {
+            dob: {
+              low: new Date("2020-01-05").toISOString(),
+              high: new Date("2020-01-10").toISOString(),
+            },
+          },
+        },
+        expected: [rows[1]],
+      },
+    ]
+
+    it.each(dateSearchTests)(
+      `should be able to run query: $query`,
+      async ({ query, expected }) => {
+        // TODO(samwho): most of these work for SQS, but not all. Fix 'em.
+        if (isSqs) {
+          return
+        }
```

Comment:
I initially tried to `!isSqs && it.each(...` a few lines up but then Jest complains about having a `beforeEach` in a `describe` with no tests, so this felt slightly less icky than using `!isSqs` in 2 places.